tools/kagami/src/swarm.rs

```diff
@@ -0,0 +1,1314 @@
+use std::{
+    collections::BTreeSet,
+    fs::File,
+    io::Write,
+    num::NonZeroU16,
+    ops::Deref,
+    path::{Path, PathBuf},
+};
+
+use color_eyre::{
+    eyre::{eyre, Context, ContextCompat},
+    Result,
+};
+use iroha_data_model::prelude::PeerId;
+use path_absolutize::Absolutize;
+use serialize_docker_compose::{DockerCompose, DockerComposeService, ServiceSource};
+use ui::UserInterface;
+
+use super::{ClapArgs, Outcome};
+
+const GIT_REVISION: &str = env!("VERGEN_GIT_SHA");
+const GIT_ORIGIN: &str = "https://github.com/hyperledger/iroha.git";
+/// Config directory that is generated in the output directory
+const DIR_CONFIG: &str = "config";
+/// Config directory inside of the docker image
+const DIR_CONFIG_IN_DOCKER: &str = "/config";
+const DIR_CLONE: &str = "iroha-cloned";
+const FILE_VALIDATOR: &str = "validator.wasm";
+const FILE_CONFIG: &str = "config.json";
+const FILE_GENESIS: &str = "genesis.json";
+const FILE_COMPOSE: &str = "docker-compose.yml";
+const DIR_FORCE_SUGGESTION: &str =
+    "You can pass `--outdir-force` flag to remove the directory without prompting";
+const GENESIS_KEYPAIR_SEED: &[u8; 7] = b"genesis";
+
+#[derive(ClapArgs, Debug)]
+pub struct Args {
+    #[command(flatten)]
+    source: ImageSourceArgs,
+    /// How many peers to generate within the docker-compose.
+    #[arg(long, short)]
+    peers: NonZeroU16,
+    /// Target directory where to place generated files.
+    ///
+    /// If the directory is not empty, Kagami will prompt it's re-creation. If the TTY is not
+    /// interactive, Kagami will stop execution with non-zero exit code. In order to re-create
+    /// the directory anyway, pass `--outdir-force` flag.
+    #[arg(long)]
+    outdir: PathBuf,
+    /// Re-create the target directory if it already exists.
+    #[arg(long)]
+    outdir_force: bool,
+    /// Do not create default configuration in the `<outdir>/config` directory.
+    ///
+    /// Default `config.json`, `genesis.json` and `validator.wasm` are generated and put into
+    /// the `<outdir>/config` directory. That directory is specified in the Docker Compose
+    /// `volumes` field.
+    ///
+    /// If you don't need the defaults, you could set this flag. The `config` directory will be
+    /// created anyway, but you should put the necessary configuration there by yourself.
```

Comment:
Not a strong preference, but I feel it's better to stick to an impersonal style in the docs:
```suggestion
    /// Setting this flag prevents copying of default configuration files into the output folder. The `config` directory will still be
    /// created, but the necessary configuration should be put there by the user manually.
```